lib/puppet/parser/functions/noop.rb

```diff
@@ -1,16 +1,16 @@
-# Set noop to true as default for current and children scopes
-Puppet::Parser::Functions::newfunction(:noop, :doc => "Set noop default to true for all resources
+# Set noop to true as default for current and children scopes, or 'reset' an inherited default with noop(false)
+Puppet::Parser::Functions::newfunction(:noop, :doc => "Set noop default for all resources
   in local scope and children scopes. This can be overriden in
   child scopes, or explicitly on each resource.
   ") do |args|
-
+      @noop_value = true if (@noop_value = args[0]).nil? or (![true, false].include? @noop_value)
```

Comment:
Let me add a little more clarity.

The specific part of the line I think needs to be reworked is `(![true, false].include? @noop_value)` which essentially says that when the value matches either boolean true or false, result in false. However according to the behavior you've stated, boolean true should not be resulting in the same behavior as boolean false.

The whole line could be cleaned up further by simplifying the expression.

`@noop_value = args[0].nil? or !!args[0]`